keras_cv/layers/preprocessing/random_resized_crop.py

```diff
@@ -0,0 +1,184 @@
+# Copyright 2022 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import warnings
+
+import tensorflow as tf
+
+from keras_cv import core
+from keras_cv.layers.preprocessing.base_image_augmentation_layer import (
+    BaseImageAugmentationLayer, )
+from keras_cv.utils import preprocessing
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class RandomResizedCrop(BaseImageAugmentationLayer):
+    """Randomly crops a part of an image and resizes it to provided size.
+
+    This implementation takes an intuitive approach, where we crop the images to a
+    random height and width, and then resize them. To do this, we first sample a
+    random value for area using `area_factor` and a value for aspect ratio using
+    `aspect_ratio_factor`. Further we get the new height and width by
+    dividing and multiplying the old height and width by the random area
+    respectively. We then sample offsets for height and width and clip them such
+    that the cropped area does not exceed image boundaries. Finally we do the
+    actual cropping operation and resize the image to `target_size`.
+
+    Args:
+        target_size: A tuple of two integers used as the target size to ultimately crop
+            images to.
+        area_factor: (Optional) A tuple of two floats, a single float or
+            `keras_cv.FactorSampler`. The ratio of area of the cropped part to
+            that of original image is sampled using this factor. Represents the
+            lower and upper bounds for the area relative to the original image
+            of the cropped image before resizing it to `target_size`.
+            `target_size`.  Defaults to (0.08, 1.0).
```

Comment:
The more I think the more I think we don't want a default for this.

Also, lets rename this to `crop_area_factor`.  We can recommendd 0.08 and 1.0, but a crop area of 0.08 is actually really aggressive for classification and might not be a sensible default there as it is for self supervised learning.

Reply:
@LukeWood In that case do we want to have a less aggressive number there? I had used 0.25 for training the smaller RegNets (200 MFLOP variant) and it seemed to work fine.

Reply:
At the time I had experimented with other values as well and found 0.25 to be good for that case.

Reply:
I don't think we want a default for this because I don't think there will be a generic best value for this.

Reply:
Got it, makes sense. 

@sayakpaul WDYT?

Reply:
@LukeWood has an interesting PoV here. I honestly haven't ever thought of default values at this depth. Luke seems to have a point. 

But let's please do one thing -- write a comprehensive guide around this layer when it's out detailing the thought process and suggest some recommended values for different tasks. The reason why I'm stressing about this is that RRC is one of the most influential layers in the domain so we'd want to ensure our users have a good guide on using it. 

WDYT? @AdityaKane2001 @LukeWood

Reply:
Sounds good. 

I think this guide can be based upon seminal literature in vision. That will give a good idea about good defaults IMO.